baseapp/baseapp_reports/graphql/object_types.py

```diff
@@ -7,27 +7,56 @@
 
 from baseapp_core.graphql import DjangoObjectType, get_object_type_for_model
 
+from .filters import ReportTypeFilter
+
 Report = swapper.load_model("baseapp_reports", "Report")
+ReportType = swapper.load_model("baseapp_reports", "ReportType")
+
 
-ReportTypesEnum = graphene.Enum.from_enum(Report.ReportTypes)
+class ReportsCount(graphene.ObjectType):
+    counts = graphene.JSONString()
 
 
-def create_object_type_from_enum(name, enum):
-    fields = {}
-    for reaction_type in enum:
-        fields[reaction_type.name] = graphene.Int()
-    fields["total"] = graphene.Int()
-    return type(name, (graphene.ObjectType,), fields)
+class BaseReportTypeObjectType:
```

Comment:
have you tried anything like:

```python
class ReportsInterface(relay.Node):
    reports_count = graphene.Field(get_reports_count_type)
```

where `get_reports_count_type` callable would handle the dynamic type creation like `create_object_type_from_enum` was handling but from the model instead?

Reply:
I did try a number of different ways to do it dynamically, but none worked as intended, some failing due to the initialization order, but most just not adding the fields as expected.